test/e2e/e2e.go

```diff
@@ -1846,7 +1848,71 @@ var _ = ginkgo.Describe("e2e ingress traffic validation", func() {
 				}
 			}
 		})
+		// This test validates ingress traffic to nodeports with externalTrafficPolicy Set to local.
+		// It creates a nodeport service on both udp and tcp, and creates a backend pod on a single node.
+		// The backend pod is using the agnhost - netexec command which replies to commands
+		// with different protocols. We use the "hostname" command to have each backend pod to reply
+		// with its hostname.
+		// We use an external container to poke the service exposed on the node and ensure that only the
+		// nodeport on the node with the backend actually receives traffic.
+		// In case of dual stack enabled cluster, we iterate over all the nodes ips and try to hit the
+		// endpoints from both each node's ips.
+		// TODO: Add Test that verifies SNAT=GWRouterIP of ingress traffic does not occur with externalTrafficPolicy=local
+		ginkgo.It("Should be allowed to node local endpoints by nodeport services with externalTrafficPolicy=local", func() {
+			serviceName := "nodeportsvclocal"
+			ginkgo.By("Creating the nodeport service")
+			// Put endpoints only on one node 
+			localNode := nodes.Items[0]
+			framework.Logf("The only node with an endpoint is: %s", localNode.Name)
+			localEndpintsSelector := map[string]string{"hostname": localNode.Name}
```

Comment:
An alternative approach would be to leave the current selector as is and ensure that hitting each node retrieves only the pod local to the node (nodename-ep).
Having one endpoint per node but narrowing it with an additional selector is a bit more convoluted.

Reply:
I agree this is a bit convoluted....However there is a corner case with the suggested approach, even if ETP feature wasn't working the test could still pass if each Noedport LB just _Happened_ to resolve to the local backend

Reply:
Granted It would be pretty rare... And over the span of "MaxTries" I would imagine the LB would try more than a single backend